### PR TITLE
separate openrave independent part query_grasp

### PR DESCRIPTION
**Pull request by fmessmer**
part of https://github.com/ipa320/cob_manipulation/issues/125

@ipa-rmb @ipa-rmb-pz FYI

### CODE REVIEW

**Comment by fmessmer** [comment]
@ipa-rmb @ipa-rmb-pz
I added a `show_grasps_rviz` action_server that uses rviz for visualization of the grasps - in addition to `show_grasps_or` that used OpenRAVE...beside querying the grasps you can now also **visualize** the grasps without having OpenRAVE installed...this should significantly help debugging the grasptables which are passed to `cob_pick_place_action` and MoveIt!

It might not be perfect yet, but it already shows a few issues (see the TODOs in the server) that might be the problem why MoveIt! is unable to find a plan for picking up the object:
 - it complains about invalid quaternion for all grasp_poses of the sdhx...
 - there again is the issue with "side-encoded" joint names, i.e. `gripper_1_joint` vs. `gripper_left_1_joint`
 - currently you cannot view the grasps for the sdh due to the mimic joint

Still, if you continue using this tools...it should be of significant help as you do not need to install OpenRAVE at all!

**Comment by fmessmer** [comment]
I fixed the handling of mimic joints and the quaternion normalization
couldn't leave it like this....I was too curious myself...:stuck_out_tongue_winking_eye:

now you clearly see that the problem with grasping with the sdhx is the grasp_poses...this can easily be fixed by modifying the csv-files!

grasp_pose_sdhx_pringles


sdh grasptables look fine:
grasp_pose_sdh_hotpot



By the way, you run the visualization like so:
 - `roslaunch cob_grasp_generation show_grasp_rviz.launch gripper:=sdh` (or `sdhx`)
 - `rosrun cob_grasp_generation show_grasps_rviz_client.py` (then enter object_name and gripper_type)

**Comment by ipa-rmb** [comment]
Awesome, thanks for looking into this. I am looking forward to see this in operation by my student.

File: cob_grasp_generation/scripts/cob_grasp_generation_action_server.py [hunk 1]
```diff
@@ -1,123 +0,0 @@
-#!/usr/bin/env python
```
Comment (ipa-rmb) [inline cob_grasp_generation/scripts/cob_grasp_generation_action_server.py:1]: Why did you remove this file? Was it obsolete?

**Review by ipa-rmb — DISMISSED** [review]

File: cob_grasp_generation/scripts/cob_grasp_generation_action_server.py (hunk 1 above)
Comment (fmessmer) [reply cob_grasp_generation/scripts/cob_grasp_generation_action_server.py:1]: nothing is lost...i created separate nodes for each action...

Comment (ipa-rmb) [reply cob_grasp_generation/scripts/cob_grasp_generation_action_server.py:1]: Aha, excellent. Thanks.

**Comment by fmessmer** [comment]
@ipa-rmb @ipa-rmb-pz did you do any further tests/enhancements with/ontopof this?
what's the current status?

@ipa-bfb @ipa-aep  FYI

**Comment by ipa-rmb-pz** [comment]
Sorry for the delay. I have just seen the Email, because I only work at Thursday and Friday.
Until now i haven't tried to run it. because there was a Demo last month, so i needed to prepare for it. since this month i will begin to try it.
i have already learned something about 'Moveit!'.

have a wonderful day! ^_^
Zhiqiang
________________________________
